app/src/main/java/com/personalchef/mealplan/MainActivity.java

```diff
@@ -182,6 +189,39 @@ public void stopForegroundService(View view) {
             }
         }catch (Exception ex) {}
     }
+    public  void populateProgress(){
+
+        // Retrieve Views
+        TextView stepCountV = findViewById(R.id.number_of_calories);
+        ProgressBar pieChart = findViewById(R.id.stats_progressbar);
+        TextView percentV = findViewById(R.id.percent);
+        double percent;
+
+        int progress = scDetail.getProgress();
+
+        pieChart.setProgress(progress);
+
+        Intent iin= getIntent();
+        Bundle b = iin.getExtras();
+        if(b!=null)
+        {
+            String j =(String) b.get("goal");
+            stepCountV.setText(stepCount + " / " + j);
```

Comment:
@Agingalus , on start of application or coming from any other activity to MainActivty, how do you plan to get/pass **"goal"** in extras of Intent/Bundle. If it's possible, then kindly update and pass all places where we are calling MainActivity.
I think, you should collaborate with @avalencia0512 and find to access **"goal"**.